File: nim/matmul.nim
import sequtils

proc matgen(n: int): seq[seq[float64]] =
  var tmp = 1.0 / float64(n) / float64(n)
  var a = newSeqWith(n, newSeq[float64](n))
  for i in 0..n-1:
    for j in 0..n-1:
      a[i][j] = tmp * float64(i - j) * float64(i + j)
  return a

proc matmul(n: int, a: seq[seq[float64]], b: seq[seq[float64]]): seq[seq[float64]] =
  var c = newSeqWith(n, newSeq[float64](n))
  for i in 0..n-1:
    for k in 0..n-1:
      var aik = a[i][k]
      var bk = b[k]
      for j in 0..n-1:
        c[i][j] += aik * bk[j]
  return c

var n = 1000
var a = matgen(n)
var b = matgen(n)
var c = matmul(n, a, b)
echo c[n shr 1][n shr 1]

File: php/matmul.php
<?php
function matgen(int $n) {
	$a = array_fill(0, $n, array_fill(0, $n, 0));
	$tmp = 1.0 / $n / $n;
	for ($i = 0; $i < $n; ++$i)
		for ($j = 0; $j < $n; ++$j)
			$a[$i][$j] = $tmp * ($i - $j) * ($i + $j);
	return $a;
}
function matmul(int $n, $a, $b) {
	$c = array_fill(0, $n, array_fill(0, $n, 0));
	for ($i = 0; $i < $n; ++$i) {
		for ($k = 0; $k < $n; ++$k) {
			$aik = $a[$i][$k];
			$bk = $b[$k];
			for ($j = 0; $j < $n; ++$j)
				$c[$i][$j] += $aik * $bk[$j];
		}
	}
	return $c;
}
$n = 1500;
$a = matgen($n);
$b = matgen($n);
$c = matmul($n, $a, $b);
echo $c[$n>>1][$n>>1], "\n";
?>
